### DISCUSSION TITLE
Build trouble on Debian 12 (bookworm)

### DISCUSSION THREAD

**Discussion by dstromberg** [ghostty-org/ghostty#3640] [Q&A] [answered]
Hi.

Tried:
    1) zig master, 2024-12-27 (I now know this is too recent)
    2) zig 0.13.0 - went with this
    3) rm -rf ~/.cache/zig/
    4) sudo apt-get install libgtk-4-dev

The error currently looks like:
```
Build Summary: 94/97 steps succeeded; 1 failed (disable with --summary none)
install transitive failure
└─ install ghostty transitive failure
   └─ zig build-exe ghostty ReleaseFast native failure
error: the following build command failed with exit code 1:
/home/dstromberg/src/ghostty/src/.zig-cache/o/992c46f4571cd9b2e5f7478e7ac53872/build /usr/local/zig-0.13.0/bin/zig /home/dstromberg/src/ghostty/src /home/dstromberg/src/ghostty/src/.zig-cache /home/dstromberg/.cache/zig --seed 0x40098274 -Z7a0ee2b79721aeec -Doptimize=ReleaseFast
```

I'm building with:
`zig build -Doptimize=ReleaseFast
`

Any suggestions?  I'm eager to try Ghostty out.

Thanks!

**Comment by rhodes-b** [comment]
Take a look here I think you might be missing a few of the required dependencies

https://ghostty.org/docs/install/build#dependencies

(below are the package but you should have `pkg-config` as well like the docs mention)
`sudo apt install libgtk-4-dev libadwaita-1-dev git`

**Comment by dariogriffo** [comment]
I'm trying to build in debian testing and getting the folllowing error:

```
/usr/local/zig/lib/libcxx/include/__mbstate_t.h:40:12: error: 'bits/alltypes.h' file not found
#  include <bits/alltypes.h>
```
The file is 
`/usr/include/x86_64-linux-musl/bits/alltypes.h`
as part of musl-dev (which installed via apt)
how do I make zig to include these files?

**Reply by dariogriffo** [reply]
@mlaventure I saw you have ghostty working in bookworm, any advice on this. I'm an absolute newbie to zig

**Reply by mlaventure** [reply]
My system has a bunch of extra toolchains installed for my daily work so I couldn't build either. I ended up building it from within a Docker container based on bookworm instead. That gave my a clean environment.

Using a container, I simply followed the readme instructions no more no less

**Reply by dariogriffo** [reply]
thank you so much!

**Reply by shuhaowu** [reply]
Do you know which toolchain is causing the issues for you? I reproduce this alltypes error on my system but can't figure out what's causing the interference.

**Reply by dariogriffo** [reply]
I can't remember. I managed to create the package and now I'm hosting public packages 
Here is the build process if you want to take a look. There is also instructions if you want to use my package how to add the apt repo
https://github.com/dariogriffo/ghostty-debian

**Comment by dstromberg** [comment]
I got ghostty 1.0 to build on Debian, but I'm having problems with it.

On one of my Debian systems, it fails to run altogether.

On the other, it takes a very long time to come up.

**Reply by dariogriffo** [reply]
any tips on build?

**Reply by yankcrime** [reply]
If you're on testing or unstable you'll need to be aware of this: https://github.com/ghostty-org/ghostty/discussions/3311#discussioncomment-11683874

**Reply by dariogriffo** [reply]
I'll try on bookworm later next week. I only have a testing laptop with me and can't build

**Comment by dstromberg** [comment, marked as answer]
This is what produced a binary for me:
Use zig 0.13.0
rm -rf ~/.cache/zig/
sudo apt-get install libgtk-4-dev
sudo apt install libadwaita-1-dev
zig build -Doptimize=ReleaseFast
